cache/blobs.go

```diff
@@ -114,6 +131,7 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 				descr, err = sr.cm.Differ.Compare(ctx, lower, upper,
 					diff.WithMediaType(mediaType),
 					diff.WithReference(sr.ID()),
+					diff.WithCompression(compressor),
 				)
```

Comment:
@tonistiigi Added this API to the fork of containerd (https://github.com/ktock/containerd/commit/11b462234b10ce7e3c028a5e54501f3d3542cf68). Now this patch doesn't create 2 blobs.  I'll work on upstreaming this API.